utilities/blob_db/blob_db_impl.cc

```diff
@@ -0,0 +1,2112 @@
+}
+
+std::pair<bool, int64_t> BlobDBImpl::fsyncFiles(bool aborted) {
+  if (aborted) return std::make_pair(false, -1);
+
+  std::vector<std::shared_ptr<BlobFile>> process_files;
+  {
+    ReadLock l(&mutex_);
+    for (auto fitr : open_blob_files_) {
+      if (fitr->NeedsFsync(true, bdb_options_.bytes_per_sync))
+        process_files.push_back(fitr);
+    }
+
+    for (auto fitr : open_simple_files_) {
+      if (fitr->NeedsFsync(true, bdb_options_.bytes_per_sync))
+        process_files.push_back(fitr);
+    }
+  }
+
+  for (auto fitr : process_files) {
+    if (fitr->NeedsFsync(true, bdb_options_.bytes_per_sync)) fitr->Fsync();
+  }
+
+  bool expected = true;
+  if (dir_change_.compare_exchange_weak(expected, false)) dir_ent_->Fsync();
+
+  return std::make_pair(true, -1);
+}
+
+std::pair<bool, int64_t> BlobDBImpl::reclaimOpenFiles(bool aborted) {
+  if (aborted) return std::make_pair(false, -1);
+
+  if (open_file_count_.load() < bdb_options_.open_files_trigger)
+    return std::make_pair(true, -1);
+
+  // in the future, we should sort by last_access_
+  // instead of closing every file
+  ReadLock l(&mutex_);
+  for (auto const& ent : blob_files_) {
+    auto bfile = ent.second;
+    if (bfile->last_access_.load() == -1) continue;
+
+    WriteLock lockbfile_w(&bfile->mutex_);
+    closeRandomAccess_locked(bfile);
+  }
+
+  return std::make_pair(true, -1);
+}
+
+std::pair<bool, int64_t> BlobDBImpl::waStats(bool aborted) {
+  if (aborted) return std::make_pair(false, -1);
+
+  WriteLock wl(&mutex_);
+
+  if (all_periods_write_.size() < bdb_options_.wa_num_stats_periods) {
+    total_periods_write_ -= (*all_periods_write_.begin());
+    total_periods_ampl_ = (*all_periods_ampl_.begin());
+
+    all_periods_write_.pop_front();
+    all_periods_ampl_.pop_front();
+  }
+
+  uint64_t val1 = last_period_write_.load();
+  uint64_t val2 = last_period_ampl_.load();
+
+  all_periods_write_.push_back(val1);
+  all_periods_ampl_.push_back(val2);
+
+  last_period_write_ = 0;
+  last_period_ampl_ = 0;
+
+  total_periods_write_ += val1;
+  total_periods_ampl_ += val2;
+
+  return std::make_pair(true, -1);
+}
+
+////////////////////////////////////////////////////////////////////////////////
+// iterate over the blobs sequentially and check if the blob sequence number
+// is the latest. If it is the latest, preserve it, otherwise delete it
+// if it is TTL based, and the TTL has expired, then
+// we can blow the entity if the key is still the latest or the Key is not
+// found
+// WHAT HAPPENS IF THE KEY HAS BEEN OVERRIDEN. Then we can drop the blob
+// without doing anything if the earliest snapshot is not
+// referring to that sequence number, i.e. it is later than the sequence number
+// of the new key
+//
+// if it is not TTL based, then we can blow the key if the key has been
+// DELETED in the LSM
+////////////////////////////////////////////////////////////////////////////////
+Status BlobDBImpl::writeBatchOfDeleteKeys(
```

Comment:
Functions start with capital letter. Please go through the change and fix all of them.